src/bridge/mod.rs

```diff
@@ -119,8 +126,11 @@ impl<IS: AsyncRead + AsyncWrite + 'static + Send> Bridge<IS> {
             IrcCommand::Join { channel } => {
                 info!(self.ctx.logger, "Joining channel"; "channel" => channel.clone());
 
+                // TODO this might need to use ::new() instead but that requires the rand feature
+                let room = RoomId::try_from(channel.clone()).unwrap();
```

Comment:
Room IDs are basically random identifiers (`!98yxkcjn1298:matrix.org`). If `channel` is the human-readable channel name, this should be a `RoomAliasId` and be constructed roughly as `RoomAliasId::try_from(format!("#{}:{}", channel, homeserver))`.

Reply:
After inspecting the rooms displayed on IRC it appears that there is a mixture of IDs and aliases (probably an issue in itself) so i swapped this to `RoomIdOrAliasId::try_from`